services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/FileBasedDeviceBackend.java

```diff
@@ -186,7 +194,66 @@ public final void get(final String tenantId, final String type, final String aut
     public void get(final String tenantId, final String type, final String authId, final JsonObject clientContext,
             final Span span,
             final Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler) {
-        credentialsService.get(tenantId, type, authId, clientContext, span, resultHandler);
+        credentialsService.get(tenantId, type, authId, clientContext, span, ar -> {
+            if (ar.result().getStatus() == HttpURLConnection.HTTP_NOT_FOUND
```

Comment:
Yes, but the code in this section only uses successful futures, while errors are encoded in the payload (e.g. 'CredentialsResult'). But anyway, I can add the check to secure this.

Reply:
Since the `CredentialsService` interface does not explicitly state that the handler will never be invoked with a failed `AsyncResult`, I think you should. This will make the code more robust against erroneous implementations anyway.